projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -215,6 +224,10 @@ export class DatagridComponent<R> implements OnInit {
 
     @ViewChild(ClrDatagridFilter, { static: false }) numericFilter: ClrDatagridFilter;
 
+    @ViewChild(ClrDatagrid, { static: true }) datagrid: ClrDatagrid;
+
+    @Input() trackBy: TrackByFunction<R> = (index, unit) => (unit && (unit as any).href ? (unit as any).href : index);
```

Comment:
Let's please explain what index and unit are. Is unit a record? Can we add return and parameter types?

Shouldn't unit be `record: R`?

Remember that `a ? a : b` is the same as `a | b` so this could be simplified to 


`return unit && (unit as any).href || index`

Consider creating a type that you can use to avoid using any

```typescript
interface HasHref {href?: string}
/**
 * Every input needs a comment
 */
@Input() trackBy: TrackByFunction<R & HasHref> = (index, record) => {
   return record && record.href || index;
};